include/dali/core/mm/memory.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_MM_MEMORY_H_
+#define DALI_CORE_MM_MEMORY_H_
+
+#include <memory>
+#include "dali/core/mm/default_resources.h"
+
+namespace dali {
+namespace mm {
+
+static const cudaStream_t host_sync = ((cudaStream_t)42);  // NOLINT
```

Comment:
Can we get stream with this value by accident?

Reply:
not really - it's not divisible by 4 and it's a pointer to a structure containing pointers, so even on 32-bit machine this would be misaligned.

Reply:
How about adding such comment in the code `that this is the is a impossible value because of ...`